include/oneapi/dpl/internal/version_impl.h

```diff
@@ -0,0 +1,40 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_VERSION_IMPL_H
+#define _ONEDPL_VERSION_IMPL_H
+
+#if _ONEDPL___cplusplus >= 202002L && __has_include(<version>)
+#    include <version> // The standard C++20 header
+#    define _ONEDPL_STD_FEATURE_MACROS_PRESENT 1
+#    define _ONEDPL_CPP20_RANGES_PRESENT ((__cpp_lib_ranges >= 201911L) && !(__clang__ && __clang_major__ < 16))
+#else
+#    define _ONEDPL_STD_FEATURE_MACROS_PRESENT 0
+#    define _ONEDPL_CPP20_RANGES_PRESENT 0
+#endif
+
+// The library version
+#define ONEDPL_VERSION_MAJOR 2022
+#define ONEDPL_VERSION_MINOR 7
+#define ONEDPL_VERSION_PATCH 0
+
+#ifndef _PSTL_VERSION
+#    define _PSTL_VERSION 14000
+#    define _PSTL_VERSION_MAJOR (_PSTL_VERSION / 1000)
+#    define _PSTL_VERSION_MINOR ((_PSTL_VERSION % 1000) / 10)
+#    define _PSTL_VERSION_PATCH (_PSTL_VERSION % 10)
+#endif
+
+// Feature macros
+#define ONEDPL_HAS_RANDOM_NUMBERS         202409L
```

Comment:
I didn't see the use of `ONEDPL_HAS_RANDOM_NUMBERS` in the code.
Is it ok?

Reply:
Should we have some test for check `ONEDPL_HAS_RANDOM_NUMBERS` state?

Reply:
This is a good question. I will think what would appropriate for testing.

Reply:
Added a few static assertions.